support-frontend/assets/helpers/productPrice/billingPeriods.ts

```diff
@@ -1,37 +1,44 @@
-const Annual = 'Annual';
-const Monthly = 'Monthly';
-const Quarterly = 'Quarterly';
+import type { ContributionType } from 'helpers/contributions';
 
-export type BillingPeriod = typeof Annual | typeof Monthly | typeof Quarterly;
+export const Annual = 'Annual';
+export const Monthly = 'Monthly';
+export const Quarterly = 'Quarterly';
+export const OneTime = 'One_Off';
 
-export type WeeklyBillingPeriod =
+export type BillingPeriod = RegularBillingPeriod | OneTimeBillingPeriod;
+export type RegularBillingPeriod =
+	| typeof Annual
 	| typeof Monthly
-	| typeof Quarterly
-	| typeof Annual;
+	| typeof Quarterly;
+type OneTimeBillingPeriod = typeof OneTime;
 
-const weeklyBillingPeriods = (): WeeklyBillingPeriod[] => {
-	return [Monthly, Quarterly, Annual];
-};
-
-const weeklyGiftBillingPeriods: WeeklyBillingPeriod[] = [Quarterly, Annual];
+export const weeklyBillingPeriods: RegularBillingPeriod[] = [
+	Monthly,
+	Quarterly,
+	Annual,
+];
+export const weeklyGiftBillingPeriods: RegularBillingPeriod[] = [
+	Quarterly,
+	Annual,
+];
 
-function billingPeriodNoun(
+export function billingPeriodNoun(
```

Comment:
these functions should probably have the `get` prefix as they represent actions
```suggestion
export function getBillingPeriodNoun(
```

Reply:
Yes, have renamed this to `getBillingPeriodNoun` and another to `getBillingPeriodTitle`